beacon-chain/graffiti/graffiti-proposal-brief.md

```diff
@@ -0,0 +1,267 @@
+
+| Component | Source | How Retrieved |
+|-----------|--------|---------------|
+| **EL Code + Commit** | Execution client (Geth/etc) | Runtime JSON-RPC call: `engine_getClientVersionV1` returns `{"code":"GE", "commit":"0x168d..."}` |
+| **CL Code + Commit** | Prysm binary | Build-time: Bazel injects git commit via ldflags → `version.GetCommitPrefix()` returns "63af" |
+
+### Component Relationships
+```
+Blockchain Service (contains and delegates)
+    │
+    ├─► EngineVersionCache (caches EL version, TTL = 6 epochs)
+    │
+    └─► Graffiti Service (resolution logic)
+            │
+            ├─► Uses: EngineClient (makes JSON-RPC calls)
+            └─► Uses: EngineVersionCache (avoids repeated calls)
+
+Pattern: Blockchain Service implements GraffitiResolver interface,
+         delegates actual work to Graffiti Service
+```
+
+## Architecture Overview
+
+### Component Flow (Simple)
+
+```
+┌─────────────────┐
+│ Validator Client│
+│  (unchanged)    │
+└────────┬────────┘
+         │ gRPC: GetBeaconBlock(graffiti="")
+         │ "I need a block to sign"
+         ↓
+┌──────────────────────────────────────┐
+│         Beacon Node                  │
+│                                      │
+│  1. RPC Handler                      │
+│     → Receives VC graffiti           │
+│                                      │
+│  2. Graffiti Service [NEW]           │
+│     → Resolves priority order        │
+│        (VC > auto > default)         │
+│                                      │
+│  3. Engine Version Cache [NEW]       │
+│     → Stores EL version info         │
+│        (TTL: 6 epochs, ~38 min)      │
+│                                      │
+│  4. Background Refresh [NEW]         │
+│     → Pre-warms cache                │
+│        (Every 2 epochs, ~13 min)     │
+│                                      │
+│  5. Engine Client Extension [NEW]    │
+│     → engine_getClientVersionV1      │
+└──────────────┬───────────────────────┘
+               │ JSON-RPC
+               ↓
+      ┌────────────────┐
+      │ Geth/Nethermind│
+      └────────────────┘
+```
+
+**How it works**:
+1. **Validator Client** calls beacon node to get a block (existing behavior, unchanged)
+2. **Beacon Node RPC** receives the request with VC's graffiti (empty if not set)
+3. **Graffiti Service** decides which graffiti to use based on priority
+4. **Cache** provides pre-fetched EL version (if needed for auto-generation)
+5. **Block returned** to VC with resolved graffiti
+
+### New Components
+
+| Component | One-Line Summary |
+|-----------|-----------------|
+| **GraffitiResolver Interface** | Interface that defines `ResolveGraffiti(vcGraffiti)` method for decoupling RPC layer from implementation |
+| **Graffiti Service** | Resolves graffiti based on priority: VC graffiti > auto-generated > default |
+| **Engine Version Cache** | Thread-safe cache storing EL version with 6-epoch TTL to avoid RPC latency |
```

Comment:
You do not need a TTL. Just update the value whenever you can and use the last good value. If the call has never succeeded then the EL abbreviation will be blank. When you generate the client string you can check if the EL abbreviation is set, and if it's not, generate the graffiti with the CL abbrev + commit only.